### PR TITLE
feat(acquiring): DOMA-11848 allow return encrypted fields unencrypted on create

### PR DESCRIPTION
**Pull request by AleX83Xpert**
## Summary by CodeRabbit

* **New Features**
  * Encryption manager gains an option to optionally return plain-text values on creation.
  * Encrypted field resolver can expose plain text on create when that option is enabled.

* **Refactor**
  * Webhook-secret handling now uses the encryption manager/resolver; direct plain-text propagation removed.
  * Secret-generation API simplified to accept a single input object.

* **Behavior**
  * Secrets generated when a webhook URL is added/changed and cleared when removed.

* **Tests**
  * Updated and added tests covering plain-text-on-create, secret generation, and edge cases.

### CODE REVIEW

File: packages/keystone/fields/EncryptedText/Implementation.js [hunk 1]
```diff
@@ -49,6 +49,32 @@ class EncryptedTextImplementation extends Text.implementation {
...
+                    && this.encryptionManager.isEncrypted(value)) {
+                    // Check if this is a create mutation (starts with 'create')
+                    const mutationName = info.operation.selectionSet.selections[0]?.name?.value || ''
+                    if (mutationName.startsWith('create')) {
+                        return this.encryptionManager.decrypt(value)
+                    }
+                }
+                
+                return value
+            },
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply packages/keystone/fields/EncryptedText/Implementation.js:97]: Trying to fix in the next commits

File: packages/keystone/fields/EncryptedText/Implementation.js [hunk 2]
```diff
@@ -49,6 +49,51 @@ class EncryptedTextImplementation extends Text.implementation {
...
 
+    gqlOutputFieldResolvers () {
+        return {
+            [this.path]: (item, args, context, info) => {
+                const value = item[this.path]
+                
+                // If returnPlainTextOnCreate is enabled and this is a create mutation,
+                // return the decrypted value. Keystone reads from DB after write,
+                // so we detect create by checking if the parent operation is a create mutation
+                if (this.encryptionManager.returnPlainTextOnCreate
```
Comment (SavelevMatthew) [inline packages/keystone/fields/EncryptedText/Implementation.js:80]: Can't we access something like existingItem from here?
Or can we use graphqlCtx to avoid repeating GQL parsing

File: packages/keystone/fields/EncryptedText/Implementation.js [hunk 3]
```diff
@@ -49,6 +49,51 @@ class EncryptedTextImplementation extends Text.implementation {
...
+                    const rootFieldKey = currentPath ? currentPath.key : null
+                    
+                    if (rootFieldKey) {
+                        // Find the matching selection in the operation's selection set
+                        // Check both alias (if present) and name to handle aliased queries
+                        const matchingSelection = info.operation.selectionSet.selections.find(
+                            selection => {
+                                const aliasMatch = selection.alias && selection.alias.value === rootFieldKey
+                                const nameMatch = selection.name && selection.name.value === rootFieldKey
+                                return aliasMatch || nameMatch
```
Comment (SavelevMatthew) [inline packages/keystone/fields/EncryptedText/Implementation.js]: That's even worse security issue....

Image

File: packages/keystone/fields/EncryptedText/Implementation.js (hunk 2 above)
Comment (AleX83Xpert) [reply packages/keystone/fields/EncryptedText/Implementation.js:80]: Refactored. Trying to use `afterChange` hook to detect the operation.

File: apps/condo/domains/acquiring/schema/fields/paymentChangeWebhook.js [hunk 4]
```diff
@@ -48,68 +55,30 @@ async function isWebhookUrlInWhitelist (url) {
...
-        // the plain text version to the API consumer so they can verify webhook signatures
-        if (context && context.req) {
-            context.req._plainWebhookSecret = plainTextSecret
-        }
-    } else if ((newCallbackUrl === null || newCallbackUrl === undefined) && existingCallbackUrl) {
-        // Clear secret when callback URL is explicitly removed
-        resolvedData['paymentStatusChangeWebhookSecret'] = null
+    // Clear secret if URL is being removed
+    if (paymentStatusChangeWebhookUrl === null || paymentStatusChangeWebhookUrl === '') {
+        resolvedData.paymentStatusChangeWebhookSecret = null
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply apps/condo/domains/acquiring/schema/fields/paymentChangeWebhook.js]: Check again

File: packages/keystone/fields/EncryptedText/Implementation.js [hunk 5]
```diff
@@ -47,10 +48,57 @@ class EncryptedTextImplementation extends Text.implementation {
...
+                    
+                    if (operation === 'create') {
+                        return this.encryptionManager.decrypt(value)
+                    }
+                }
+                
+                return value
+            },
+        }
     }
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply packages/keystone/fields/EncryptedText/Implementation.js:99]: You are good rabbit.
Fixed :)

**Comment by pahaz** [comment]
Please add the task ID

**Comment by AleX83Xpert** [comment]
> Please add the task ID

Done

File: packages/keystone/fields/EncryptedText/Implementation.js [hunk 6]
```diff
@@ -47,10 +48,57 @@ class EncryptedTextImplementation extends Text.implementation {
...
+                    && context._encryptedTextOperations) {
+                    
+                    // Check if this item was just created by looking up the operation in context
+                    // Use per-field key to support multiple EncryptedText fields in the same model
+                    const itemId = item.id
+                    const fieldPath = this.path
+                    const operation = context._encryptedTextOperations[`${this.listKey}:${itemId}:${fieldPath}`]
+                    
+                    if (operation === 'create') {
+                        return this.encryptionManager.decrypt(value)
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply packages/keystone/fields/EncryptedText/Implementation.js:92]: fixed

File: packages/keystone/fields/EncryptedText/Implementation.js [hunk 7]
```diff
@@ -47,10 +48,57 @@ class EncryptedTextImplementation extends Text.implementation {
...
+        if (!this._needsOperationTracking) return
+        
+        // Store operation type in context for use in gqlOutputFieldResolvers
+        // Use per-field key to avoid collisions when a model has multiple EncryptedText fields
+        if (!context._encryptedTextOperations) {
+            context._encryptedTextOperations = {}
+        }
+        const key = `${listKey}:${updatedItem.id}:${this.path}`
+        context._encryptedTextOperations[key] = operation
+    }
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply packages/keystone/fields/EncryptedText/Implementation.js:71]: Should be fixed in the last commit
